superset-frontend/src/SqlLab/constants.ts

```diff
@@ -65,3 +65,6 @@ export const SQL_FUNCTIONS_AUTOCOMPLETE_SCORE = 90;
 export const SCHEMA_AUTOCOMPLETE_SCORE = 60;
 export const TABLE_AUTOCOMPLETE_SCORE = 55;
 export const COLUMN_AUTOCOMPLETE_SCORE = 50;
+
+// LocalStorage keys
+export const SQL_LAB_AUTOCOMPLETE_SETTING_KEY = 'isSQLLabAutocompleteEnabled';
```

Comment:
Would it be better if we keep all local storage keys as an `enum` and move into `localStorageHelpers.js`? The keys are global so it makes sense to keep them in a centralized place.

The values can even be typed:

```ts
/**
 * Local storage manager
 */

export function safeJsonParse(jsonString: string | null) {
  if (jsonString === null) return null;
  try {
    return JSON.parse(jsonString);
  } catch {
    return null;
  }
}

export type StorageValues = {
  key1: Type1,
  key2: Type2,
};

export type StorageKey = keyof StorageValues;

export const defaultStorageValues: StorageValues = {
   key1: value1,
   key2: value2
};

export function entries() {
  Object.values(Storage).forEach(key => {
    return [key, getItem(key)];
  });
}

export function getItem<K extends StorageKey>(key: K): StorageValues[K] {
  return safeJsonParse(localStorage.getItem(key)) ?? defaultStorageValues[key];
}

export function setItem(key: StorageKey, value: string | number | object) {
  return localStorage.setItem(key, JSON.stringify(value));
}
```

Reply:
yeah, that was my original thought, but i realized the existing keys that are used don't do that yet. maybe i'll make a PR first to centralize all the keys, then i'll rebase this one on it

Reply:
I'm going to implement this now, but probably won't add default values here, if only because i think sometimes the caller might want to se a different default